packages/user-ui/src/pages/projects/_id/components/comments.rs

```diff
@@ -1,8 +1,243 @@
+use by_components::icons::{
+    arrows::{ChevronDown, ChevronUp},
+    chat::{SquareChat, SquareMark},
+    emoji::ThumbsUp,
+    internet_script::Code,
+    links_share::Link2,
+    photos::AddPhoto,
+};
 use dioxus::prelude::*;
+use dioxus_translate::{translate, Language};
+
+use crate::{
+    components::icons::{bold::BoldIcon, more::MoreIcon},
+    pages::projects::_id::components::i18n::CommentTranslate,
+};
 
 #[component]
-pub fn Comment() -> Element {
+pub fn Comment(lang: Language) -> Element {
+    let mut comment = use_signal(|| "".to_string());
+    let mut comments = use_signal(|| Vec::<String>::new());
+    let tr: CommentTranslate = translate(&lang);
+    let mut add_comment = move || {
+        if !comment().is_empty() {
+            comments.push(comment().clone());
+            comment.set("".to_string());
+        }
+    };
+
     rsx! {
-        div { "comment" }
+        div { class: "max-w-[1300px] h-[48px] mb-[75px]",
+            // comment counts
+            div { class: "w-full h-[24px] mt-[40px] mb-[10px] flex flex-row justify-start items-center gap-[8px]",
+                div {
+                    SquareChat { color: "#555462" }
+                }
+                // TODO: connect to comment numbers
+                p { "{comments.len()}" }
+            }
+
+            // text write area
+            div { class: "max-w-[1300px] min-h-[48px] relative border-[1px] border-[#B4B4B4] rounded-[8px] flex justify-start items-center pl-[12px] gap-[8px]",
+                SquareChat { color: "#8095EA" }
+                // text input area
+                textarea {
+                    class: "w-full h-[48px] py-[12px] font-semibold text-[15px] leading-normal outline-none",
+                    placeholder: "{tr.reply_box_text}",
+                    value: "{comment}",
+                    oninput: move |e| comment.set(e.value().clone()),
+                    onkeypress: move |e| {
+                        if e.key() == Key::Enter {
+                            e.prevent_default();
+                            add_comment();
+                        }
+                    },
+                }
+            }
+
+            //comments
+            div { class: "w-full h-auto flex flex-col justify-center items-start",
+                for c in comments.iter() {
+                    // Each comment is now handled independently by a separate component
+                    CommentItem { lang, comment: c.clone() }
+                }
+            }
+        }
+    }
+}
+
+#[component]
+pub fn CommentItem(lang: Language, comment: String) -> Element {
+    let mut show_comments = use_signal(|| false);
+    let mut show_reply_input = use_signal(|| false);
+    let mut reply = use_signal(|| "".to_string());
+    let mut replies = use_signal(|| Vec::<String>::new());
+    let tr: CommentTranslate = translate(&lang);
```

Comment:
```suggestion
    let tr: CommentItemTranslate = translate(&lang);
```
별도의 translate 를 사용해주세요.